### PR TITLE
Big http download upgrade#26717

### PR DESCRIPTION
**Pull request by KozhinovAlexander**
PR for Big HTTP Download - Upgrade #26717 Issue

### CODE REVIEW

**Comment by pfalcon** [comment]
Please update PR title, so it was clear what this PR is about.

File: samples/net/sockets/big_http_download/README.rst [hunk 1]
```diff
@@ -36,6 +36,11 @@ Build the Zephyr version of the application like this:
    :goals: build
    :compact:
 
+.. warning::
+
+   On some boards the stack size CONFIG_MAIN_STACK_SIZE must be increased!
```
Comment (pfalcon) [inline samples/net/sockets/big_http_download/README.rst:41]: This is not an ok warning to put in the docs. If the stack size needs to be increased, let's increase it to safe enough value. (The fact that Zephyr stack size requirements in Zephyr is constantly grows is a known matter, I raised concern of it, but nobody seems to pay attention.)

File: samples/net/sockets/big_http_download/prj.conf [hunk 2]
```diff
@@ -3,6 +3,7 @@ CONFIG_NEWLIB_LIBC=y
 CONFIG_MBEDTLS=y
 CONFIG_MBEDTLS_ENABLE_HEAP=y
 CONFIG_MAIN_STACK_SIZE=2176
+CONFIG_SAMPLE_BIG_HTTP_DL_NUM_ITER=3
```
Comment (pfalcon) [inline samples/net/sockets/big_http_download/prj.conf:6]: Please don't change existing settings without good motivation. Such changes and motivation should go into a separate PR (on its own). Current usecase for big_http_download is that a user can start it and use for assessing IP stack overall stability, e.g. wait and watch until it downloads 1GB total data (then one kinda assume that "IP stack in Zephyr works").

File: samples/net/sockets/big_http_download/src/big_http_download.c [hunk 3]
```diff
@@ -4,6 +4,10 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <logging/log.h>
```
Comment (pfalcon) [inline samples/net/sockets/big_http_download/src/big_http_download.c:7]: big_http_download is a POSIX-compatible sample (i.e. written with purely POSIX API, and portable to any POSIX system). Please run `make -f Makefile.posix` in the sample dir to ensure it remains so. (This change likely breaks that.)

File: samples/net/sockets/big_http_download/src/big_http_download.c [hunk 4]
```diff
@@ -33,6 +37,11 @@
 
 #endif
 
+#define KiB	*1024
+#define MiB	*1024 KiB
+#define Bytes2KiB(Bytes)	(((float)Bytes)/(1.0f KiB))
```
Comment (pfalcon) [inline samples/net/sockets/big_http_download/src/big_http_download.c:42]: I'm not sure about usage of floating-point arithmetic. This would be one cause of stack usage growth you faced. And it would grow binary size, and potentially make it not fit on smaller devices, which would be a loss of functionality.

**Review by pfalcon — CHANGES_REQUESTED** [review]
Thanks for the patches. They definitely make this sample more complicated, but I'm not sure they *improve* it. Please keep in mind that this a **sample**, not a wget implementation or something. Its purpose is to show that it's possible to write simple, POSIX-compatible applications for Zephyr. Specifically:

1. This sample should be simple for user to study (the less code the better, or at least "more code" is not an improvement).
2. This sample should be simple, and small, and thus run on as many device as possible (usage of floating-point arithmetic is specifically avoided).
3. This sample should be buildable on a standard POSIX system (e.g. Linux) using `make -f Makefile.posix`, and this should be achieved with as few #ifdef's in the source code as possible (so, adding more isn't a good direction). 

I think that this patch in its current shape deteriorates the sample on all the points above.


Looking just at the functionality, I agree that showing download speed would be useful. As I mentioned elsewhere, the reason I didn't add it myself is, well, because the figures which come out aren't exactly to brag about :-I. And then, there're multiple download speeds - e.g. averaged over all iterations, averaged over the current iteration, instantaneous (i.e. averaged over small period of time, 3-5s). IMHO, the most useful for Zephyr *testing* would be the last.

**Comment by Nukersson** [comment]
> Please update PR title, so it was clear what this PR is about.

New PR with updated name and branch name were created: #26734

File: samples/net/sockets/big_http_download/src/big_http_download.c (hunk 4 above)
Comment (Nukersson) [reply samples/net/sockets/big_http_download/src/big_http_download.c:42]: Changed to int arithmetics.
